wintools.py

```
@@ -1382,6 +1387,230 @@ def api_quicksetup_apply():
+    # Validate name — no newlines, null chars, or equals signs
+    if any(c in name for c in ["\n", "\r", "\0", "="]):
+        return jsonify({"success": False, "error": "Invalid variable name"}), 400
+
+    # Protect critical system variables from deletion
+    if action == "delete" and name.upper() in _PROTECTED_SYSTEM_VARS and scope == "system":
+        return jsonify({"success": False, "error": f"Cannot delete critical system variable '{name}'"}), 400
+
+    # Check admin for system scope
+    if scope == "system":
+        try:
+            is_admin = ctypes.windll.shell32.IsUserAnAdmin() != 0
```

Ruff (0.16.1)

[error] Undefined name `ctypes`

(F821)